Enable clearing of last error message in CLI

markcmiller86 opened the pull request:
### Description

The CLI method, `GetLastError()` returns the last error message. But, there was no way to clear it out to `""`. So, once an error happened, it would forever be some string until a new error happened. There is also an associated `errorFlag` in the supporting class and there was already a `ClearError()` method in that class but it only reset the `errorFlag` and left the `lastError` unchanged.

I made `ClearError()` also reset `lastError` to `""` and then added an optional integer arg to `GetLastError(int-clr)` so that it can accept an integer argument, `int-clr`. If `int-clr` is a non-zero integer, then after obtaining whatever string is in `lastError`, it will also reset `lastError` to `""`. If `int-clr` is zero, it behaves the same as `GetLastError()`. Any other argument will cause `GetLastError()` to fail.

<!-- Please include a summary of the change and any other information and instructions to help reviewers understand the work -->

<!-- Note that all the checklist items in various bulleted lists below end with ~~ characters.
     This is to facilitate striking them out when they do not apply.
     One just has to add ~~ characters just before the opening square bracket [ -->

### Type of change

<!-- Please check one of the boxes below -->

* ~~[ ] Bug fix~~
* [x] New feature~~
* ~~[ ] Documentation update~~
* ~~[ ] Other~~ <!-- please explain with a note below -->

### How Has This Been Tested?

<!-- Please describe the tests you've added or any tests that already cover this change. Include relevant information, such as which operating system you tested on. -->

### Reminders:

- Please follow the [style guidelines][1] of this project.
- Please perform a self-review of your code before submitting a PR and asking others to review it.
- Please assign reviewers (see [VisIt's PR procedures][2] for more information).

### Checklist:

<!-- For items in this checklist that do not apply, simply insert two tilde chars, `~~`, just ahead of the left bracket char, `[` at the beginning of a line. Each line ends with two tilde chars to make doing such ~~strikeouts~~ easy. -->

- [x] I have commented my code where applicable.~~
- [x] I have updated the release notes.~~
- [x] I have made corresponding changes to the documentation.~~
- ~~[ ] I have added debugging support to my changes.~~
- [x] I have added tests that prove my fix is effective or that my feature works.~~
- ~~[ ] I have confirmed new and existing unit tests pass locally with my changes.~~
- ~~[ ] I have added new baselines for any new tests to the repo.~~
- [x] I have NOT made any changes to [*protocol* or *public interfaces*][3] in an RC branch.~~

[1]: https://visit-sphinx-github-user-manual.readthedocs.io/en/develop/dev_manual/StyleGuide.html
[2]: https://visit-sphinx-github-user-manual.readthedocs.io/en/develop/dev_manual/pr_create.html#reviewers
[3]: https://visit-sphinx-github-user-manual.readthedocs.io/en/develop/dev_manual/RCDevelopment.html#communication-protocols-and-public-apis

markcmiller86 committed:
enable clearing of last err msg

markcmiller86 committed:
update relnotes

markcmiller86 committed:
add test for clearing error

markcmiller86 committed:
add missing ;

markcmiller86 committed:
Merge branch 'develop' into feat-mcm86-06dec22-clr-cli-error

biagas commented on src/doc/python_scripting/functions.rst:
```diff
@@ -3654,16 +3654,19 @@ GetLastError
     GetLastError returns a string containing the last error message that VisIt
-    issued.
+    issued since being cleared. If int-clr arg, if present, is non-zero, once
+    the message is retrieved, it is also cleared.
```
```suggestion
    issued since being cleared. If int-clr is present and is non-zero, once
    the message is retrieved it is also cleared.
```
Here and lines 3668-9 as well.

markcmiller86 committed:
Update src/doc/python_scripting/functions.rst

Co-authored-by: Kathleen Biagas <biagas2@llnl.gov>

markcmiller86 committed:
Merge branch 'develop' into feat-mcm86-06dec22-clr-cli-error

biagas commented on src/doc/python_scripting/functions.rst:
```diff
@@ -3654,16 +3654,19 @@ GetLastError
 
     The GetLastError function returns a string containing the last error
-    message that VisIt issued.
+    message that VisIt issued since being cleared. If int-clr arg, if present,
+    is non-zero, once the message is retrieved, it is also cleared.
```
this sections should match lines 3661-2.

markcmiller86 replied:
Whoops. Ok, fixed that one too. Also, adjusted to current policy of one sent. per line. Previously, we couldn't do that here because these strings were taken verbatim into Python doc strings. But, newest doc string generation actually does the wrapping now so I also adjusted changes paragraphs to sent. per line.

markcmiller86 committed:
Update section on GetLastError

Newest python doc string code gen wraps lines so we can maintain one sent. per line policy here now too.
